pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1592,21 +1611,73 @@ func (n *NetworkProvider) getNonOverlappingDefaultCIDR(ctx context.Context) (*ne
 			}
 			// increment potential IP
 			potentialDefaultIP = incrementIPForDefaultCIDR(potentialDefaultIP)
+			//check for overlap
+			// if it overlaps continue
+			if foundClusterNet {
+				if clusterNet.Contains(defaultNet.IP) || defaultNet.Contains(clusterNet.IP) {
+					continue
+				}
+
+			}
+
+			if foundPodNet {
+				if podCIDR.Contains(defaultNet.IP) || defaultNet.Contains(podCIDR.IP) {
+					continue
+				}
+
+			}
+
+			if foundServiceNet {
+				if serviceCIDR.Contains(defaultNet.IP) || defaultNet.Contains(serviceCIDR.IP) {
+					continue
+				}
 
-			// check for overlap
-			// if it overlaps continue with the loop
-			if clusterNet.Contains(defaultNet.IP) || defaultNet.Contains(clusterNet.IP) {
-				continue
 			}
+
 			// return the first available option that does not overlap
 			return defaultNet, nil
 		}
 	}
+
 	// if the loop finishes, it means that we have gone through all available cidr blocks in the available ranges in aws
 	// return an error that cro was unable to find an option
 	return nil, errorUtil.New("could not find a default cidr block")
 }
 
+func (n *NetworkProvider) getPodCIDR(networkConf *v12.Network) (*net.IPNet, bool, error) {
```

Comment:
we don't need to target `NetworkProvider` for these functions, as we don't use it.